### PR TITLE
Lavaland Russians + More Megafauna

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Russians are invading lavaland. They feed plasma to a portal to get more russians.

The russians will have to complete a series of objectives leading up to a horde mode spawn/defense phase, followed by a boss fight, followed by getting their macguffin.

Two new bosses, bubblegum and the colossus, are included. Bubblegum is mostly done, the colossus is mostly unstarted.

### CODE REVIEW

**Comment by Razharas** [comment]
USE PROPER ICONS ITS VERY INPORTANT

**Comment by KorPhaeron** [comment]
I WILL

**Comment by pudl** [comment]
can you link the last pr or copy paste the description

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	idle_vision_range = 5
+	loot = list(/obj/structure/closet/crate/necropolis/dragon)
+	butcher_results = list(/obj/item/weapon/ore/diamond = 5, /obj/item/stack/sheet/sinew = 5, /obj/item/stack/sheet/animalhide/ashdrake = 10, /obj/item/stack/sheet/bone = 30)
+
+	deathmessage = "crashes to the ground."
+	death_sound = 'sound/magic/demon_dies.ogg'
+	damage_coeff = list(BRUTE = 1, BURN = 0.5, TOX = 1, CLONE = 1, STAMINA = 0, OXY = 1)
+
+
+/obj/effect/at_shield
```
Comment (Steelpoint) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: I see what you did there.

**Comment by Steelpoint** [comment]
You'll have to further explain the whole concept of the Russian Invasion thing (inb4 removes people from station argument). 

I've also seen a Bubblegum in action, its pretty cool but still needs balancing. 

Looking forward to seeing what happens.

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/mob/living/simple_animal/hostile/megafauna/colossus
+	name = "colossus"
+	desc = "get in the fucking robot."
+	health = 2500
+	maxHealth = 2500
+	attacktext = "chomps"
+	attack_sound = 'sound/magic/demon_attack1.ogg'
+	icon_state = "eva"
+	icon_living = "eva"
+	icon_dead = "dragon_dead"
```
Comment (Dorsisdwarf) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: Is this intentional?

Comment (Steelpoint) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: He did says he's barely started on this mob.

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "at_shield2"
+	layer = 6.1
+	luminosity = 2
+	var/target
+
+/obj/effect/at_shield/New()
+	..()
+	spawn()
+		for(var/i in 1 to 80-1)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: or just ```0 to 80```

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: also this is kinda bad maybe just have it orbit the robot

Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: Can I do an orbit so slow it doesnt actually turn? I was considering it

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: With 0 rotation segments, it will stay still but still follow it's target.

**Comment by Bawhoppen** [comment]
ORIGINAL ASSETS KOR PLEASE YOU EVEN HAVE AN AUSOPS

File: code/modules/mining/lavaland/ruins/russian_invasion.dm [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+		points += P.amount
+		user << "You feed [P.amount] sheets into the machine. It now has [points] sheets of plasma fueling it."
+		qdel(P)
+
+/obj/structure/russian_portal/proc/select_reinforcement(mob/user)
+	var/selection = input(user, "Request Reinforcements", "Portal") as null|anything in list("Prisoner (4)", "Conscript (8)", "Spetsnaz (20)", "Commissar (20)", "Spare Ammo (10)", "Medical Supplies (10)")
+	if(!selection || !Adjacent(user))
+		return
+	switch(selection)
+		if("Prisoner (4)")
```
Comment (Iamgoofball) [inline code/modules/mining/lavaland/ruins/russian_invasion.dm]: d a t u m i z e

**Comment by KorPhaeron** [comment]
@Bawhoppen 

What is not original here?

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm (hunk 3 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: That works, thanks

**Comment by Bawhoppen** [comment]
walls, floor, equipment, guns, pretty much everything because nobody ever cares about shit

**Comment by KorPhaeron** [comment]
I'm not gonna design and equip an entire new faction when we have the assets for an unused one sitting in the code already

**Comment by Bawhoppen** [comment]
b-b-b-bullshit

**Comment by MMMiracles** [comment]
bawhoppen if you want original assets then why don't you contribute to the project instead of complaining.

oh wait.

**Comment by Zilenan91** [comment]
>The russians will have to complete a series of objectives leading up to a horde mode spawn/defense phase, followed by a boss fight, followed by getting their macguffin.

What do you mean by horde mode spawn/defense phase?

**Comment by Bawhoppen** [comment]
are you literally autistic
I contribute all the time but theres kind of a fucking freeze going on??????

**Comment by MMMiracles** [comment]
>bitch about something not 'using original assets'

>wont actually help but is still content on bitching about it

whew

**Comment by Bawhoppen** [comment]
you are literally mentally deficient

**Comment by Steelpoint** [comment]
It would be nice to see a sort of neo-Russian look for the Russians but I also think using the pre-existing Russian sprites (which are never really used) is perfectly fine.

The sprites can always be changed after the fact.

**Comment by Iamgoofball** [comment]
@Steelpoint time capsule USSR soldiers is funnier

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	death_sound = 'sound/magic/demon_dies.ogg'
+	damage_coeff = list(BRUTE = 1, BURN = 0.5, TOX = 1, CLONE = 1, STAMINA = 0, OXY = 1)
+
+
+/obj/effect/at_shield
+	name = "anti-toolbox field"
+	desc = "A shimmering forcefield protecting the colossus."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "at_shield2"
+	layer = 6.1
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: Use `FLY_LAYER` or `ABOVE_ALL_MOB_LAYER`

Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm]: Sure

**Comment by Bawhoppen** [comment]
:ru: :ru: :ru: :ru: :ru: :rage: :us: :us: :us: :us: :us: 

:ru: :ru: :ru: :ru: :ru: :gun: :us: :us: :us: :us: :us: 

:ru: :ru: :ru: :ru: :ru: :bomb: :us: :us: :us: :us: :us: 

:ru: :skull: :ru: :ru: :skull:  :bomb: :us: :skull: :skull: :skull: :us: 

:skull: :skull: :skull: :skull: :skull:  :crown: :us: :skull: :skull: :skull: :skull: 

:us: :skull: :us: :us: :gun: :cn: :cn: :cn: :skull: :cn: :cn: :cn: :cn: :cn: :cn: :cn: :cn: :cn: :cn: 

:skull: :skull: :us: :us: :bomb: :cn: :cn: :cn: :skull: :skull: :cn: :cn: :cn: :cn: :cn: :cn: :cn: :cn: :cn: 

:skull: :skull: :skull: :us: :rocket: :cn: :cn: :cn: :skull: :skull: :cn: :cn: :cn: :cn: :cn: :cn: :cn: :cn: :cn:

:us: :us: :us: :us: :hash::one:  :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull:

**Comment by Bawhoppen** [comment]
what happened to my comment from last night

**Comment by KorPhaeron** [comment]
Doing this in multiple prs now

**Pull request closed without merging**